tools/generate-code.py

```diff
@@ -26,13 +26,15 @@ def generate_towire_field(field, allfields, lang):
 
     if isinstance(field.fieldtype, pyln.proto.message.array_types.ArrayType):
         if lang == 'js':
-            print('    for (let v in value[_n]) {{\n'
+            print('    for (let v of value) {{\n'
                   '        buf = Buffer.concat([buf, towire_{ftype}(v)]);\n'
+                  '        _n++;\n' #Should be here?..
```

Comment:
This change is wrong, but here's why.

We are now using _n to count the total *number* of fields we output towire, so we can simply assert that we output everything in the dictionary: this catches if they put unknown fields in the dict (e.g. a typo!).  So we want to increment n once for every field we output, not for every element in an array we might output!